e2etest/bgptests/bgp.go

```diff
@@ -1209,6 +1209,185 @@ var _ = ginkgo.Describe("BGP", func() {
+
+			for _, c := range FRRContainers {
+				validateFRRPeeredWithNodes(cs, c, ipFamily)
+			}
+
+			allNodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+			framework.ExpectNoError(err)
+
+			var pools []metallbv1beta1.IPAddressPool
+			var services []*corev1.Service
+			for i := 0; i < 2; i++ {
+				ginkgo.By(fmt.Sprintf("configure addresspool number %d", i+1))
+				firstIP, err := config.GetIPFromRangeByIndex(addressRange, i*10+1)
+				framework.ExpectNoError(err)
+				lastIP, err := config.GetIPFromRangeByIndex(addressRange, i*10+10)
+				framework.ExpectNoError(err)
+				addressesRange := fmt.Sprintf("%s-%s", firstIP, lastIP)
+
+				pool := metallbv1beta1.IPAddressPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: fmt.Sprintf("test-addresspool%d", i+1),
+					},
+					Spec: metallbv1beta1.IPAddressPoolSpec{
+						Addresses: []string{addressesRange},
+					},
+				}
+				pools = append(pools, pool)
+
+				resources := metallbconfig.ClusterResources{
+					Pools: pools,
+					Peers: metallb.PeersForContainers(FRRContainers, ipFamily),
+					BGPAdvs: []metallbv1beta1.BGPAdvertisement{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "test-advertisement1",
+							},
+						},
+					},
+				}
+				err = ConfigUpdater.Update(resources)
+				framework.ExpectNoError(err)
+
+				ginkgo.By(fmt.Sprintf("configure service number %d", i+1))
+				svc, _ := testservice.CreateWithBackend(cs, f.Namespace.Name, fmt.Sprintf("svc%d", i+1), testservice.TrafficPolicyCluster, func(svc *corev1.Service) {
+					svc.Annotations = map[string]string{"metallb.universe.tf/address-pool": fmt.Sprintf("test-addresspool%d", i+1)}
+				})
+				defer testservice.Delete(cs, svc)
+
+				for _, c := range FRRContainers {
+					ginkgo.By(fmt.Sprintf("checking connectivity of service %d to external frr container %s", i+1, c.Name))
+					validateService(cs, svc, allNodes.Items, c)
+				}
+
+				services = append(services, svc)
+			}
+
+			var bgpAdvs []metallbv1beta1.BGPAdvertisement
+			for i := 0; i < 2; i++ {
+				ginkgo.By(fmt.Sprintf("setting peer selector for addresspool number %d to peer %s", i+1, testContainers[i].Name))
+				peer := ""
+				for _, p := range resources.Peers {
+					if strings.Contains(p.Name, testContainers[i].Name) {
+						peer = p.Name
+					}
+				}
+				bgpAdv := metallbv1beta1.BGPAdvertisement{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: fmt.Sprintf("test-advertisement%d", i+1),
+					},
+					Spec: metallbv1beta1.BGPAdvertisementSpec{
+						IPAddressPools: []string{fmt.Sprintf("test-addresspool%d", i+1)},
+						Peers:          []string{peer},
+					},
+				}
+				bgpAdvs = append(bgpAdvs, bgpAdv)
+
+				resources := metallbconfig.ClusterResources{
+					Pools:   pools,
+					Peers:   metallb.PeersForContainers(FRRContainers, ipFamily),
+					BGPAdvs: bgpAdvs,
+				}
+				err = ConfigUpdater.Update(resources)
+				framework.ExpectNoError(err)
+
+				for _, c := range FRRContainers {
```

Comment:
See my suggestoin above to pick named peers.
For the notAdvertised, I would add a function that filters FRRContainers out of a container and iterate over those.